services/src/handlers/datasets.rs

```diff
@@ -1087,15 +1125,16 @@ mod tests {
         ))
         .unwrap();
 
-        if let MetaDataDefinition::OgrMetaData(meta_data) = &mut meta_data {
+        if let crate::datasets::storage::MetaDataDefinition::OgrMetaData(meta_data) = &mut meta_data
```

Comment:
shouldn't this be the MetaDataDefinition from services/api/model ?

Reply:
I thought of `auto_detect_meta_data_definition` more of like an internal utility function which isn't directly connected to the API. Because of that I used internal types.